File: frontend/src/lib/api/commands.ts
import { jsonFetch } from './core';

export type CommandKind = 'command' | 'skill';
export type CommandScope = 'user' | 'project' | 'plugin';

export type CommandEntry = {
  slug: string;
  description: string;
  kind: CommandKind;
  scope: CommandScope;
  source_path: string;
};

export type CommandsList = {
  entries: CommandEntry[];
};

/** Fetch the palette of slash commands + skills. Pass the session's
 *  `working_dir` as `cwd` so any project-local `.claude/commands` are
 *  included with highest precedence. */
export function listCommands(
  cwd: string | null | undefined,
  fetchImpl: typeof fetch = fetch
): Promise<CommandsList> {
  const query = cwd ? `?cwd=${encodeURIComponent(cwd)}` : '';
  return jsonFetch<CommandsList>(fetchImpl, `/api/commands${query}`);
}

File: src/bearings/api/models/commands.py
"""Slash-command palette DTOs (commands + skills surfaced in the
textarea popup)."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class CommandOut(BaseModel):
    """One entry in the slash-command palette (command or skill).

    `slug` is the token inserted into the textarea without the leading
    `/` — it matches what the Claude Code CLI accepts (e.g. `fad:ship`,
    `pr-review-toolkit:review-pr`). `scope` records where the entry came
    from so the UI can group or badge it; `source_path` is kept for
    debugging only — the client should not display it.
    """

    slug: str
    description: str
    kind: Literal["command", "skill"]
    scope: Literal["user", "project", "plugin"]
    source_path: str


class CommandsListOut(BaseModel):
    entries: list[CommandOut]
